contracts/WildcardSteward_v2.sol

```diff
@@ -277,16 +302,19 @@ contract WildcardSteward_v2 is Initializable {
                         .div(patronageOwedByTokenPatron)
                 )
             );
-            amountToMint = (newTimeLastCollected.sub(previousTokenCollection))
-                .mul(tokenGenerationRate[tokenId]);
-        } else {
-            amountToMint = now.sub(timeLastCollected[tokenId]).mul(
-                tokenGenerationRate[tokenId]
+            timeSinceLastMint = (
+                newTimeLastCollected.sub(previousTokenCollection)
             );
+        } else {
+            timeSinceLastMint = now.sub(timeLastCollected[tokenId]);
 
         }
-        erc20Minter[tokenId].mint(currentOwner, amountToMint);
-        emit CollectLoyalty(tokenId, currentOwner, amountToMint);
+        mintManager.tokenMint(
+            currentOwner,
+            timeSinceLastMint,
+            tokenGenerationRate[tokenId]
+        );
+        emit CollectLoyalty(tokenId, currentOwner, timeSinceLastMint);
```

Comment:
This event should include the tokenGenerationRate for the token maybe. Then you can calculate how many tokens were minted just from the event. Not critical I guess since that info is easy to get and static.